CoreFunctionality/test/unit/src/au/gov/asd/tac/constellation/functionality/ConstellationErrorManagerNGTest.java

```diff
@@ -0,0 +1,139 @@
+
+import au.gov.asd.tac.constellation.views.errorreport.ErrorReportDialogManager;
+import au.gov.asd.tac.constellation.views.errorreport.ErrorReportEntry;
+import au.gov.asd.tac.constellation.views.errorreport.ErrorReportSessionData;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import static org.mockito.Mockito.mock;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+import org.testng.annotations.Test;
+
+/**
+ * Testing for the ConstellationErrorManager class
+ * @author OrionsGuardian
+ */
+public class ConstellationErrorManagerNGTest {
+    private static final Logger LOGGER = Logger.getLogger(ConstellationErrorManagerNGTest.class.getName());
+    
+    @Test
+    public void runExceptionsTest() {
+        final MockedStatic<ErrorReportDialogManager> erdmStatic = Mockito.mockStatic(ErrorReportDialogManager.class);
+        final ErrorReportDialogManager erdm = mock(ErrorReportDialogManager.class);
+        erdmStatic.when(ErrorReportDialogManager::getInstance).thenReturn(erdm);
+
+        final ConstellationErrorManager errorManager = new ConstellationErrorManager();
+        LOGGER.addHandler(errorManager);
+        LOGGER.setLevel(Level.ALL);
+        
+        // generate exception log messages
+        simulateException(Level.SEVERE);
+        simulateException(Level.WARNING);
+        simulateException(Level.INFO);
+        simulateException(Level.FINE);
+
+        LOGGER.log(Level.INFO, "\n ------- start wait for data ... {0}", new Date());
+        boolean dataAvailable = waitForDataToBeAvailable(Level.SEVERE, 1);
+        assertTrue(dataAvailable);
+        LOGGER.log(Level.INFO, "\n ------- finished waiting for data ... {0}", new Date());
+        
+        // confirm the exceptions have been stored in the SessionData class
+        confirmExceptionStored(Level.SEVERE, 1);
+        confirmExceptionStored(Level.WARNING, 1);
+        confirmExceptionStored(Level.INFO, 1);
+        confirmExceptionStored(Level.FINE, 1);
+        
+        confirmEntryOccurrences(Level.SEVERE, 1);
+        
+        // NOTE: the following 2 calls NEED to be on the SAME LINE to generate identical stack traces
+        simulateException(Level.SEVERE); simulateException(Level.SEVERE);
+        
+        LOGGER.log(Level.INFO, "\n ------- 2.start wait for data ... {0}", new Date());
+        dataAvailable = waitForDataToBeAvailable(Level.SEVERE, 2);
+        assertTrue(dataAvailable);
+        LOGGER.log(Level.INFO, "\n ------- 2.finished waiting for data ... {0}", new Date());
+
+        // confirm correct number of entries and occurrences
+        confirmExceptionStored(Level.SEVERE, 2);
+        confirmEntryOccurrences(Level.SEVERE, 2);
+    }
+    
+    public void simulateException(final Level logLevel){
```

Comment:
just wondering if some of these helper functions can be made private? Just because we shouldn't be using them elsewhere